iot_packet_verifier/src/balances.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{burner::Burn, pdas};
+use anchor_lang::AccountDeserialize;
+use data_credits::DelegatedDataCreditsV0;
+use futures_util::StreamExt;
+use helium_crypto::PublicKey;
+use solana_client::{client_error::ClientError, nonblocking::rpc_client::RpcClient};
+use solana_sdk::program_pack::Pack;
+use solana_sdk::pubkey::Pubkey;
+use sqlx::{Pool, Postgres};
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+pub struct Balances {
+    pub provider: Arc<RpcClient>,
+    pub balances: Arc<Mutex<HashMap<PublicKey, Balance>>>,
```

Comment:
Do we need to use full public key here as the hashmap key ? PublicKeyBinary much more efficient

Reply:
Sure, PublicKeyBinary works